runelite-client/src/main/java/net/runelite/client/Notifier.java

```diff
@@ -217,19 +215,22 @@ public void processFlash(final Graphics2D graphics)
 			return;
 		}
 
-		if (Instant.now().minusMillis(MINIMUM_FLASH_DURATION_MILLIS).isAfter(flashStart))
+		final int minimum_flash_duration_millis = runeLiteConfig.flashNotificationDuration();
+		final int minimum_flash_duration_ticks = minimum_flash_duration_millis / Constants.CLIENT_TICK_LENGTH;
+
+		if (Instant.now().minusMillis(minimum_flash_duration_millis).isAfter(flashStart))
 		{
 			switch (flashNotification)
 			{
-				case FLASH_TWO_SECONDS:
-				case SOLID_TWO_SECONDS:
+				case FLASH_FOR_DURATION:
+				case SOLID_FOR_DURATION:
 					flashStart = null;
 					return;
 				case SOLID_UNTIL_CANCELLED:
 				case FLASH_UNTIL_CANCELLED:
 					// Any interaction with the client since the notification started will cancel it after the minimum duration
-					if ((client.getMouseIdleTicks() < MINIMUM_FLASH_DURATION_TICKS
-						|| client.getKeyboardIdleTicks() < MINIMUM_FLASH_DURATION_TICKS
+					if ((client.getMouseIdleTicks() < minimum_flash_duration_ticks
```

Comment:
This currently is using MINIMUM_FLASH_DURATION_MILLIS for two different things, the flash time and the minimum time before activity can cancel a notif (this is so notifs don't get triggered and then cancelled prior to you being able to see them). Since you are replacing both things with 1 new configuration value it is strange since now the min cancel time for flash_until_cancelled is actually the "flash notification duration"

It would probably make more sense to have the config be a simple off/flash/solid and then have a 2nd option for time, and then use time = 0 as until cancelled.